[Liu Wanyu] iP

wanyu-l opened the pull request.

Commit:
Add Gradle support

Commit:
-Skeletal version of the program that echoes input, unless input is "bye" of any combination of upper/lower case letters, then program terminates and does not respond to further inputs

Commit:
-changed local variable, "curr" to "currentLine"
-added documentation

Commit:
- changed approach to identifying keywords, from "toLowerCase()" to "equalsIgnoreCase"
- inputs are now treated as valid commands for adding tasks, unless it is keyword for termination
- added methods to support storing of things to be done

Commit:
- added documentation for relevant methods

Commit:
- added documentation

Commit:
- added Task class
- tasks can now be marked as completed or incomplete

Commit:
- added documentation

Commit:
- added features to handle possible wrong inputs
- modified messages displayed

Commit:
- modified program to only accept certain inputs and prompts user to key in the correct way
- added features to handle some possible wrong inputs

Commit:
- corrected for catching and handling exceptions

Commit:
- added documentation
- modified code for better flow
- tweaked error handling for invalid inputs

Commit:
- input like "done1" are no longer accepted, spaces are expected

Commit:
- modified spacing issues with printed strings

Commit:
- fixed error when calling list when it is empty

Commit:
- corrected typo

Commit:
- updated input and expected txt files

Commit:
- tweaked dialog prompts for different types of tasks

Commit:
- added feature to delete tasks
- added method to display commands to help with user input

Commit:
- added documentation

wanyu-l committed:
- corrected 'date' to 'deadline' in dialog prompt
- corrected '<Event>' to 'event' and '<Deadline>' to 'deadline' in dialog prompt
- reorganized sequence of dialog prompts
- tested some more cases

wanyu-l committed:
- changed some method implementation such that enum is part of the implementation of the program, enum 'Command' and method 'doSomething'
- added documentation
- tweaked dialog prompts

wanyu-l committed:
Add Level-7 features
Update approach to detecting user input

wanyu-l committed:
Add Level-8
Update attribute of Event and Deadline to store LocalDate or String or both

wanyu-l committed:
Merge branch 'branch-Level-7'

wanyu-l committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	.gitignore
#	src/main/java/Duke.java

wanyu-l committed:
Merge commit '526ae3c400d490f686ae925973e42530eb92642a' (Level-3 commit that was in another branch)

# Conflicts:
#	.gitignore
#	src/main/java/Duke.java

wanyu-l committed:
Update approach to identify user input
Update organization of code, split into various java files

wanyu-l committed:
Update organization of tiles by packages

wanyu-l committed:
Add various Junit tests for corresponding methods

wanyu-l committed:
Add documentation to most public methods and fields

wanyu-l committed:
Update code style to match coding standards

wanyu-l committed:
Add FindCommand
Update Parser and TaskList to support find

wanyu-l committed:
Merge branch 'A-JavaDoc'

wanyu-l committed:
Merge branch 'A-CodingStandard'

# Conflicts:
#	src/main/java/commands/ByeCommand.java
#	src/main/java/commands/Command.java
#	src/main/java/commands/DeadlineCommand.java
#	src/main/java/commands/DeleteCommand.java
#	src/main/java/commands/DoneCommand.java
#	src/main/java/commands/DueCommand.java
#	src/main/java/commands/EventCommand.java
#	src/main/java/commands/ListCommand.java
#	src/main/java/commands/ToDoCommand.java
#	src/main/java/duke/DukeException.java
#	src/main/java/parser/Parser.java
#	src/main/java/storage/Storage.java
#	src/main/java/tasks/DeadLineTask.java
#	src/main/java/tasks/EventTask.java
#	src/main/java/tasks/Task.java
#	src/main/java/tasks/ToDoTask.java
#	src/main/java/ui/Ui.java

wanyu-l committed:
Merge branch 'Level-9'

# Conflicts:
#	src/main/java/parser/Parser.java
#	src/main/java/tasks/TaskList.java

wanyu-l committed:
Update various files to match coding standards

wanyu-l committed:
Update java documentation and method names
TaskList: adding and correction of java documentation, changed method name
Task: correction of java documentation

wanyu-l committed:
Delete Ui.class

wanyu-l committed:
Delete text-ui-test/New folder directory

wanyu-l committed:
Delete src/data directory

wanyu-l committed:
Remove unnecessary files
Deleted various .class and .txt files for local tests

wanyu-l committed:
Update FIndCommand
added missing java documentation
removed isExit() method

wanyu-l committed:
Merge branch 'add-gradle-support'

wanyu-l committed:
Add Gradle to project
Updated ParserTest, TasksTest and CommandsTest for
automation of Junit Tests.
Included tests for instances of FindCommand in ParserTest
and CommandsTest.
Organized import statements in above mentioned files.

wanyu-l committed:
Update Gradle and Ui
File path for duke is updated.
Additional dialogue added for helperMessage in Ui.

wanyu-l committed:
Add Gui to Duke
Updated various java files to return String as representation
of outcome of different executed tasks.
Updated Ui to store and return standard messages.

wanyu-l committed:
Update Gradle
Added feature for Checkstyle from AB3

wanyu-l committed:
Revert "Update Gradle"

This reverts commit 9505f94be90bd4cb26e9755f42abab3cd56f0ede.

wanyu-l committed:
Revert "Revert "Update Gradle""

This reverts commit 9d52009d5b8cc0b14178e3022772e8f7d4a09e85.

wanyu-l committed:
Merge branch 'A-CheckStyle'

wanyu-l committed:
Update various files according to Checkstyle

wanyu-l committed:
Update DukeApp
A parser instance is stored instead of
creating  a new parser to parse every input.

wanyu-l committed:
Update DukeApp
removed unused class field.

wanyu-l committed:
Update README.md

wanyu-l committed:
Update README.md

wanyu-l committed:
Update README.md

wanyu-l committed:
Update README.md

wanyu-l committed:
Update README.md

wz27 commented on src/main/java/commands/ByeCommand.java:
```diff
@@ -0,0 +1,38 @@
+     *
+     * @param s the entire line of user input
+     */
+    public ByeCommand(ArrayList<String> s) {
+        super(s);
```
Should variables with single-character names be used for nested loops only?

wanyu-l replied:
changed to "userInput", thanks for pointing it out!

wz27 commented on src/main/java/commands/Command.java:
```diff
@@ -0,0 +1,52 @@
+     *
+     * @param s the entire line of user input
+     */
+    public Command(ArrayList<String> s) {
+        this.input = s;
```
Should variables with single-character names be used for nested loops only?

wanyu-l replied:
changed to "userInput", thanks for pointing it out!

wz27 commented on src/main/java/tasks/DeadLineTask.java:
```diff
@@ -0,0 +1,127 @@
+ * as per input by user.
+ */
+public final class DeadLineTask extends Task {
+    /**
+     * Stores the type of task information
+     * that identifies a DeadlineTask.
+     */
+    private final String taskType = "[D]";
```
Should Javadoc of class members be specified on a single line?

wanyu-l replied:
changed to single line, thanks for catching my mistake!

wz27 commented on src/main/java/tasks/DeadLineTask.java:
```diff
@@ -0,0 +1,127 @@
+     */
+    private final String taskType = "[D]";
+
+    /**
+     * Stores the do-by-date information.
+     */
+    private final String dueDate;
+
```
Should Javadoc of class members be specified on a single line?

wanyu-l replied:
changed to single line, thanks for catching my mistake!

wz27 commented on src/main/java/tasks/DeadLineTask.java:
```diff
@@ -0,0 +1,127 @@
+     */
+    private final String dueDate;
+
+    /**
+     * Stores the do-by-date information as a LocalDate.
+     */
+    private LocalDate localDate;
```
Should Javadoc of class members be specified on a single line?

wanyu-l replied:
changed to single line, thanks for catching my mistake!

wz27 commented on src/main/java/tasks/EventTask.java:
```diff
@@ -0,0 +1,129 @@
+ * as per input by user.
+ */
+public final class EventTask extends Task {
+    /**
+     * Stores the type of task information
+     * that identifies a EventTask.
+     */
+    private final String taskType = "[E]";
```
Should Javadoc of class members be specified on a single line?

wanyu-l replied:
changed to single line, thanks for catching my mistake!

wz27 commented on src/main/java/tasks/TaskList.java:
```diff
@@ -0,0 +1,187 @@
+import java.time.LocalDate;
+import java.util.ArrayList;
+
+public final class TaskList {
```
Should there be Javadov header comments for this class?

wanyu-l replied:
added missing java documentation for TaskList, thanks for catching it!

wz27 commented on src/main/java/tasks/TaskList.java:
```diff
@@ -0,0 +1,187 @@
+public final class TaskList {
+    private ArrayList<Task> tasks;
+
+    public TaskList(ArrayList<Task> inputTasks) {
```
Should there be Javadov header comments for this public method?

wanyu-l replied:
added missing java documentation for TaskList, thanks for catching it!

wz27 reviewed (commented):
Looks good! Some very minor coding standard changes requested.

wanyu-l committed:
Update various files for correct java documentation format and user prompts
class members are changed to single line representation in various java files
missing java documentation added in TaskList
name for parameter for constructors in subclasses of Command changed to "userInput"

wanyu-l committed:
Update various files' parameter names and Ui
* changed single character parameter names to more descriptive
words like "userInput" and "description"
*  corrected display error in Ui, where "tasks.Task" was shown instead of "Task"
*  changed method name from "getTask" to "getDescription" and relevant changes
in related files

wanyu-l committed:
Update various files
* added assert feature to all the update files in this commit except Task and ToDoTask
* reorganized code in DeleteCommand for better readability

wanyu-l committed:
Update various files code quality
* removed unnecessary "this" and "super" in subclasses of Task and Command
* changed "lst" to "list" for parameters of command classes' "execute" method
* organized code in DeleteCommand for better readability

wanyu-l committed:
Update TaskList
* changed implementation of how to find relevant tasks
in findTask and findTasksDue to use stream

wanyu-l committed:
Update DoneCommand, Task and TaskList
* added assert in DoneCommand
* changed setIsDone in Task to return String instead of printing
* corrected error in using stream in method findTasksDue in TaskList

wanyu-l committed:
Merge pull request #1 from wanyu-l/A-Assertions

Update various files
* added assert feature to all the update files in this commit except Task and ToDoTask
* reorganized code in DeleteCommand for better readability

wanyu-l committed:
Merge branch 'master' into A-CodeQuality

wanyu-l committed:
Merge pull request #2 from wanyu-l/A-CodeQuality

Update various files code quality

wanyu-l committed:
Merge branch 'master' into A-Streams

wanyu-l committed:
Merge pull request #3 from wanyu-l/A-Streams

A-streams

wanyu-l committed:
Update DueCommand
* corrected for undeclared local variable

wanyu-l committed:
Update various files
* moved Duke into a different folder
* Deadline and Event only accept standard input for date
* corrected indentation in Task

wanyu-l committed:
Add Reschedule feature
* added Reschedule command to amend date for Events and Deadlines
* removed DukeException in EventTask, DeadlineTask and Storage to correct
for problem in not being able to save list to local file
* updated DukeApp for an alert for tasks due on 'today' and included a snooze method
to Reschedule to the day after, to snooze or not is user's choice
* added java documentation for various files

wanyu-l committed:
Update README.md

wanyu-l committed:
Update README.md

wanyu-l committed:
Update README.md

wanyu-l committed:
Update README.md

wanyu-l committed:
Update README.md

wanyu-l committed:
Update README.md

wanyu-l committed:
Update README.md

wanyu-l committed:
Update various files for improvement of GUI
* changed TasksTest accordingly as DukeException is not thrown by constructor
of subclasses of Task anymore
* added DukeMascot as icon for all displayed windows
* removed line separator in Ui and affected files
* reorganized DukeApp to only store instance of Ui, TaskList, Storage and UserPrompts
for easier access
* updated DialogBox for color filled background and display of round image

wanyu-l committed:
Update author credit
* corrected for typo
* corrected section of reused code

wanyu-l committed:
Update README.md

wanyu-l committed:
Merge branch 'master' into heads/Level-10

wanyu-l committed:
Update DukeApp
* removed unused line of code

wanyu-l committed:
Update gitignore

wanyu-l committed:
Merge remote-tracking branch 'origin/heads/Level-10'

wanyu-l committed:
Update various files for code readability and optimise user interaction
* changed condition to display helper message in Ui, only shown if input is unrecognized command
* corrected typos and unintended changes of javadoc in various files during refactoring
* added javadoc to various Test files and amended them according to coding standards
* updated Ui to no longer have static methods or variables and corrected affected files accordingly
* updated various commands to be more specific to different types of unrecognized user inputs

wanyu-l committed:
Add Ui screenshot

wanyu-l committed:
Update files in Ui to rectify errors
* changed approach to get png and fxml files using getResourceAsStream instead of FileInputStream methods
and updated corresponding files
* changed shadowJar's archiveBaseName to match the name displayed by GUI, "Memo-Assistant Duke"

wanyu-l committed:
Update DueCommand and FindCommand for more accurate display message
* changed "tasks" to "task(s)" or "task", depending on context,  for both files
* added png for display of popup windows that interact with user

wanyu-l committed:
Update README.md

wanyu-l committed:
Update README.md

wanyu-l committed:
Update README.md

wanyu-l committed:
Update README.md

wanyu-l committed:
Update README.md

wanyu-l committed:
Update MainWindow' to match changes to result of executing DueCommand

wanyu-l committed:
Update README.md

wanyu-l committed:
Update README.md

wanyu-l committed:
Update README.md

wanyu-l committed:
Update README.md

wanyu-l committed:
Set theme jekyll-theme-cayman

wanyu-l committed:
Update README.md

wanyu-l committed:
Update README.md

wanyu-l committed:
Update README.md

wanyu-l committed:
Update to allow a valid time to be added
* updated DeadLineTask and EventTask to check for time when constructed
* updated Reschedule Command to add time component when relevant
* updated Task and TaskList for easier amendments to tasks
* updated TasksTest to check for time when a new DeadLineTask or EventTask is created

wanyu-l committed:
Update GUI and various files for better user interaction
* updated fxml files to resize GUI to fit smaller screens
* updated helper message in Ui for better prompts
* updated DialogBox for bigger user font size
* updated DeadlineTask and EventTask to check for valid date and time
and corresponding Command to provide better feedback

wanyu-l committed:
Update README.md

wanyu-l committed:
Update README.md

wanyu-l committed:
Update README.md

wanyu-l committed:
Update README.md
